### ISSUE TITLE
Compare two inputs

### ISSUE THREAD

**Issue by StefanMunnes** [rstudio/shinyvalidate#83]
Hello,
I'm wondering if it is easily possible to compare two input values, e.g. two email addresses are equal? I tried the following approach, but it doesn't work bc the `input$mail` is not in an observe environment:
```r
iv$add_rule("mail2", sv_equal(input$mail, "Not the same email address"))
```

If I wrap the `input$mail` into `shiny::reactive(input$mail)`, the app starts but outputs the following error:
```r
An unexpected error occurred during input validation: comparison (==) is possible only for atomic and list types
```

**Comment by sw-jakobgepp** [comment]
I used a workaround, where I created a new instance within an `observeEvent()`. This way, I can set rules depending on other inputs.

In my case, I had two date inputs (`global_datetime_start` and `global_datetime_end`), and they should not be more than a week apart.

```
  shiny::observeEvent(input$global_datetime_end), {

    iv2 <- shinyvalidate::InputValidator$new()
    iv2$add_rule("global_datetime_start", shinyvalidate::sv_required())
    iv2$add_rule(
      "global_datetime_start",
      shinyvalidate::sv_between(
        left = input$global_datetime_end - 60 * 60 * 24 * 7,
        right = input$global_datetime_end
      )
    )

    iv2$enable()
  })
```
